benchmark/sirun/runall.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+set -e
+
+DIRS=($(ls -d */ | sed 's:/$::')) # Array of subdirectories
+CWD=$(pwd)
+
+function cleanup {
+  for D in "${DIRS[@]}"; do
+    rm -f "${CWD}/${D}/meta-temp.json"
+  done
+}
+
+trap cleanup EXIT
+
+if test -f ~/.nvm/nvm.sh; then
+  source ~/.nvm/nvm.sh
+else
+  source /usr/local/nvm/nvm.sh
+fi
+
+# run each test in parallel for a given version of Node.js
+# once all of the tests have complete move on to the next version
+
+TOTAL_CPU_CORES=$(nproc 2>/dev/null || echo "24")
+
+export CPU_AFFINITY="${CPU_START_ID:-$TOTAL_CPU_CORES}" # Benchmarking Platform convention
+
+nvm install $MAJOR_VERSION # provided by each benchmark stage
+export VERSION=`nvm current`
+export ENABLE_AFFINITY=true
+
+echo "using Node.js ${VERSION}"
+CPU_AFFINITY="${CPU_START_ID:-$TOTAL_CPU_CORES}" # reset for each node.js version
+SPLITS=${SPLITS:-1}
+GROUP=${GROUP:-1}
+
+BENCH_COUNT=0
+for D in "${DIRS[@]}"; do
+  cd "${D}"
+  variants="$(node ../get-variants.js)"
+  for V in $variants; do BENCH_COUNT=$(($BENCH_COUNT+1)); done
+  cd ..
+done
+
+echo "BENCH_COUNT: ${BENCH_COUNT}"
+GROUP_SIZE=$(($(($BENCH_COUNT+$SPLITS-1))/$SPLITS)) # round up
+
+BENCH_INDEX=0
+BENCH_END=$(($GROUP_SIZE*$GROUP))
+BENCH_START=$(($BENCH_END-$GROUP_SIZE))
+
+if [[ ${GROUP_SIZE} -gt 24 ]]; then
```

Comment:
```suggestion
if [[ ${GROUP_SIZE} -gt ${TOTAL_CPU_CORES} ]]; then
```

Could we actually just use 50% of all cores as upper bound? That way the GC and JIT compilation may happen in the background and should have less influence on the results.